### ISSUE TITLE
Ticking entity: butterfly crashes server while on flax from Supplementaries mod

### ISSUE THREAD

**Issue by 4SURU4** [itsmeow/betteranimalsplus#322]
```
Description: Ticking entity

java.lang.IllegalArgumentException: Cannot set property IntegerProperty{name=age, clazz=class java.lang.Integer, values=[0, 1, 2, 3, 4, 5, 6, 7]} to 8 on Block{supplementaries:flax}, it is not an allowed value
	at net.minecraft.world.level.block.state.StateHolder.m_61124_(StateHolder.java:125) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading}
	at dev.itsmeow.betteranimalsplus.common.entity.EntityButterfly.m_8024_(EntityButterfly.java:283) ~[betteranimalsplus-1.18.2-11.0.5-forge.jar%2349!/:1.18.2-11.0.5] {re:classloading}
	at net.minecraft.world.entity.Mob.m_6140_(Mob.java:706) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8107_(LivingEntity.java:2506) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:caelus.mixins.json:MixinLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:clienttweaks.mixins.json:LivingEntityAccessor,pl:mixin:APP:expandability.mixins.json:swimming.LivingEntityMixin,pl:mixin:APP:netherportalfix.mixins.json:LivingEntityAccessor,pl:mixin:APP:supplementaries.mixins.json:LivingEntityLadderMixin,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8107_(Mob.java:495) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at net.minecraft.world.entity.AgeableMob.m_8107_(AgeableMob.java:127) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading}
	at net.minecraft.world.entity.animal.Animal.m_8107_(Animal.java:54) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading,pl:mixin:APP:quark.mixins.json:AnimalMixin,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2253) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:caelus.mixins.json:MixinLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:clienttweaks.mixins.json:LivingEntityAccessor,pl:mixin:APP:expandability.mixins.json:swimming.LivingEntityMixin,pl:mixin:APP:netherportalfix.mixins.json:LivingEntityAccessor,pl:mixin:APP:supplementaries.mixins.json:LivingEntityLadderMixin,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8119_(Mob.java:307) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at dev.itsmeow.betteranimalsplus.common.entity.EntityButterfly.m_8119_(EntityButterfly.java:122) ~[betteranimalsplus-1.18.2-11.0.5-forge.jar%2349!/:1.18.2-11.0.5] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:652) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.world.level.block.state.StateHolder.m_61124_(StateHolder.java:125) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading}
	at dev.itsmeow.betteranimalsplus.common.entity.EntityButterfly.m_8024_(EntityButterfly.java:283) ~[betteranimalsplus-1.18.2-11.0.5-forge.jar%2349!/:1.18.2-11.0.5] {re:classloading}
	at net.minecraft.world.entity.Mob.m_6140_(Mob.java:706) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8107_(LivingEntity.java:2506) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:caelus.mixins.json:MixinLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:clienttweaks.mixins.json:LivingEntityAccessor,pl:mixin:APP:expandability.mixins.json:swimming.LivingEntityMixin,pl:mixin:APP:netherportalfix.mixins.json:LivingEntityAccessor,pl:mixin:APP:supplementaries.mixins.json:LivingEntityLadderMixin,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8107_(Mob.java:495) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at net.minecraft.world.entity.AgeableMob.m_8107_(AgeableMob.java:127) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading}
	at net.minecraft.world.entity.animal.Animal.m_8107_(Animal.java:54) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,re:classloading,pl:mixin:APP:quark.mixins.json:AnimalMixin,pl:mixin:A}
	at net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2253) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:caelus.mixins.json:MixinLivingEntity,pl:mixin:APP:citadel.mixins.json:LivingEntityMixin,pl:mixin:APP:clienttweaks.mixins.json:LivingEntityAccessor,pl:mixin:APP:expandability.mixins.json:swimming.LivingEntityMixin,pl:mixin:APP:netherportalfix.mixins.json:LivingEntityAccessor,pl:mixin:APP:supplementaries.mixins.json:LivingEntityLadderMixin,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_8119_(Mob.java:307) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betteranimalsplus.mixins.json:MobAccessor,pl:mixin:A}
	at dev.itsmeow.betteranimalsplus.common.entity.EntityButterfly.m_8119_(EntityButterfly.java:122) ~[betteranimalsplus-1.18.2-11.0.5-forge.jar%2349!/:1.18.2-11.0.5] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:652) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
-- Entity being ticked --
Details:
	Entity Type: betteranimalsplus:butterfly (dev.itsmeow.betteranimalsplus.common.entity.EntityButterfly)
	Entity ID: 327
	Entity Name: Butterfly
	Entity's Exact location: -206.50, 64.50, -579.50
	Entity's Block location: World: (-207,64,-580), Section: (at 1,0,12 in -13,4,-37; chunk contains blocks -208,-64,-592 to -193,319,-577), Region: (-1,-2; contains chunks -32,-64 to -1,-33, blocks -512,-64,-1024 to -1,319,-513)
	Entity's Momentum: 0.09, -0.01, -0.07
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.world.level.Level.m_46653_(Level.java:486) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:319) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:299) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220228.144236-srg.jar%23114!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}
```

**Comment by itsmeow** [comment]
It seems Flax might not be properly interacting due to its double-height aging system. It's possible this could be a problem with the values returned by Supplementaries, but there is probably some checks that could prevent an error like this that I should've done anyway.

**Comment by 4SURU4** [comment]
It could be Butterfly + Flax + Rain combo. I needed to set Butterfly spawn to 0 and kill that (all) butterfly to get world running. Funny one

**Comment by TheVillain-Inc** [comment]
I've also got this crash, in a 'greenhouse' so no rain in the mix. 

Oddly they'd been locked in there for some times before this happened.
[crash-2022-06-13_04.56.14-server.txt](https://github.com/itsmeow/betteranimalsplus/files/8894205/crash-2022-06-13_04.56.14-server.txt)

**Comment by xinwolfthorn** [comment]
I have this same issue in 1.16.5, a butterfly got inside my house and got stuck on top of a chest, i see no flax anywhere around, though it still has almost the same error..

https://pastebin.com/SabPuNyw

here is a link to the issue i posted to the supplementaries github.. https://github.com/MehVahdJukaar/Supplementaries/issues/397

modder says they don't think the error is on their side..

**Comment by xinwolfthorn** [comment]
this problem still exists for 1.16.5 =(

**Comment by itsmeow** [comment]
Fix released for 1.16.5
